media/js/pocket/updates-signup.es6.js

```diff
@@ -0,0 +1,125 @@
+            error = form.querySelector('.error-email-invalid');
+        } else {
+            // backend error, tampered form/missing fields
+            error = form.querySelector('.error-try-again-later');
+        }
+        if (error) {
+            error.classList.remove('hidden');
+        }
+    },
+
+    handleFormSuccess: () => {
+        const formWrapper = document.getElementById(
+            'pocket-updates-form-wrapper'
+        );
+        const thanksWrapper = document.getElementById('pocket-updates-thanks');
+        formWrapper.classList.add('hidden');
+        thanksWrapper.classList.remove('hidden');
+    },
+    postToBackend: (payload, url, successCallback, errorCallback) => {
+        const xhr = new XMLHttpRequest();
+
+        xhr.onload = function (e) {
+            let response = e.target.response || e.target.responseText;
+
+            if (typeof response !== 'object') {
+                response = JSON.parse(response);
+            }
+
+            if (response) {
+                if (
+                    response.status === 'success' &&
+                    e.target.status >= 200 &&
+                    e.target.status < 300
+                ) {
+                    successCallback();
+                } else {
+                    errorCallback(response);
+                }
+            } else {
+                errorCallback();
+            }
+        };
+        xhr.onerror = errorCallback;
+        xhr.open('POST', url, true);
+        xhr.setRequestHeader('Content-type', 'application/json');
+        xhr.setRequestHeader('X-Requested-With', 'XMLHttpRequest');
+        xhr.timeout = 5000;
+        xhr.ontimeout = errorCallback;
+        xhr.responseType = 'json';
+        xhr.send(JSON.stringify(payload));
+    },
+
+    subscribe: (e) => {
+        const payload = Object.fromEntries(new FormData(form).entries());
+        const url = form.getAttribute('action');
+
+        e.preventDefault();
+        e.stopPropagation();
+
+        // Disable form fields until POST has completed.
+        FormUtils.disableFormFields(form);
+
+        // Clear any prior messages that might have been displayed.
+        FormUtils.clearFormErrors(form);
+
+        // Submit to the server
+        UpdatesForm.postToBackend(
+            payload,
+            url,
+            UpdatesForm.handleFormSuccess,
+            UpdatesForm.handleFormError
+        );
+    },
+
+    getSearchParams: () => {
+        // isolated to make easier to mock in tests
+        return new URL(window.location.href).searchParams;
+    },
+
+    populateFromQuerystring: () => {
+        const soughtParams = ['campaign', 'medium', 'source'];
+        // Grab the UTM params for the hidden form fields, if we can, set them
+        // as default form values
+        for (const [key, value] of UpdatesForm.getSearchParams()) {
+            const strippedKey = key.replace('utm_', '');
+            const allowableRegex = RegExp(/^[\w\d/.%-_]{1,100}$/);
```

Comment:
I think the `\d` is redundant here. Adding an underscore at the end also changes this to "Match a character in the range of % to _", which is not quite what we need.

The `\w` at the beginning should already match alpha numeric characters (including underscores)

Reply:
Yep, you're 100% right - my regex cheatsheet betrayed me :)